device_doctor/lib/src/process_helper.dart

```diff
@@ -67,3 +69,50 @@ Future<String> eval(String executable, List<String> arguments,
 
   return output.trimRight();
 }
+
+/// Splits [from] into lines and selects those that contain [pattern].
+Iterable<String> grep(Pattern pattern, {@required String from}) {
+  return from.split('\n').where((String line) {
+    return line.contains(pattern);
+  });
+}
+
+Future<void> killAllRunningProcessesOnWindows(String processName) async {
+  ProcessManager processManager = LocalProcessManager();
+  while (true) {
```

Comment:
What if a different user owns the process, and device doctor cannot kill it? The tool will hang here trying to do the impossible. We should limit the tries on this loop

Reply:
Set tries limit 3.